### PR TITLE
COp conflict between old section markers and new section markers

### PR DESCRIPTION
**Pull request by carriepl**
This PR does the following : 
-fixes false positives in the detection of old section markers
-add an error message specific for the case where no section marker has been detected in the code
-add an error message specific for the case where both types of section markers have been detected in the code
-adds to the section that processes the old syntax checks that were only present in the section for new syntax
-adds a few comments

### CODE REVIEW

File: theano/gof/op.py [hunk 1]
```diff
@@ -1062,12 +1062,35 @@ def load_c_code(self):
...
         self.code_sections = dict()
         for i, code in enumerate(self.func_codes):
-            if ('THEANO_APPLY_CODE_SECTION' in code or
-                'THEANO_SUPPORT_CODE_SECTION' in code):
+            if self.backward_re.search(code):
                 # This is backward compat code that will go away in a while
+
+                # Check for code outside of the supported sections
                 split = self.backward_re.split(code)
+                if split[0].strip() != '':
```
Comment (nouiz) [inline theano/gof/op.py]: Was this check done in the old code? Or this is just a port from the new code?

Other then this questions, the PR seem good to me.

File: theano/gof/op.py [hunk 2]
```diff
@@ -1062,12 +1062,35 @@ def load_c_code(self):
...
+            if self.backward_re.search(code):
+                old_markers_present = True
+            if self.section_re.search(code):
+                new_markers_present = True
+
+        if old_markers_present and new_markers_present:
+            raise ValueError('Both the new and the old syntax for '
+                             'identifying code sections are present in the '
+                             'provided C code. These two syntaxes should not '
+                             'be used at the same times')
```
Comment (lamblin) [inline theano/gof/op.py]: "time."

Comment (carriepl) [inline theano/gof/op.py]: done

File: theano/gof/op.py (hunk 1 above)
Comment (carriepl) [inline theano/gof/op.py]: Good point. I removed the check which was only a port from the new code.

**Pull request merged**